### PR TITLE
Move chaff to middleware

### PR DESCRIPTION
**Pull request by sethvargo**
**Release Note**

```release-note
Move chaff handling into middleware
```

### CODE REVIEW

**Comment by sethvargo** [comment]
https://github.com/google/exposure-notifications-server/pull/1357?w=1

**Comment by sethvargo** [comment]
https://github.com/google/exposure-notifications-server/pull/1357/files?w=1

File: internal/publish/publish.go [hunk 1]
```diff
@@ -130,14 +130,15 @@ func NewServer(ctx context.Context, cfg *Config, env *serverenv.ServerEnv) (*Ser
 	}, nil
 }
 
-func (s *Server) Routes(ctx context.Context, cfg *Config) *mux.Router {
+func (s *Server) Routes(ctx context.Context) *mux.Router {
 	logger := logging.FromContext(ctx).Named("publish")
 
 	r := mux.NewRouter()
+	r.Use(middleware.ProcessChaff(s.tracker))
```
Comment (sethvargo) [inline internal/publish/publish.go:137]: Should this come after maintenance mode and other middlewares?

**Review by mikehelmick — APPROVED** [review]

File: internal/publish/publish.go (hunk 1 above)
Comment (sethvargo) [reply internal/publish/publish.go:137]: (It was previously first, so I kept that behavior, but I'm wondering if it's the _correct_ behavior)

**Pull request merged**